bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java

```diff
@@ -136,141 +143,95 @@ private void refresh() {
                         (100 - pvLimitation.intValue()) <= config.limitationTresholdValue, config.limitationDuration);
             }
 
-            Channel channelConsumption = getThing()
-                    .getChannel(SenecHomeBindingConstants.CHANNEL_SENEC_POWER_CONSUMPTION);
-            if (channelConsumption != null) {
-                updateState(channelConsumption.getUID(),
-                        new QuantityType<Power>(
-                                getSenecValue(response.energy.homePowerConsumption).setScale(2, RoundingMode.HALF_UP),
-                                Units.WATT));
```

Comment:
I don't know if it's relevant. But what changed here is that the creation of the QuantityType instance is always done even if `channelConsumption` is null. I don't know if that could result in errors, which could happen if the value cannot be evaluated if `channelConsumption` is null and in case that dependency does exist.
If this is an actual problem you could pass the QuantityType as `Consumer`. Like: `() -> new QuantityType(...` and then in the method it can be called when the parameter is not null.

Reply:
Hey,
I don't see the problem that you mention in the code. The Data is passed by getSenecValue() from the Senec Device. So if the channel is null or not has no influence on the creation of the QuantityType, only the .getUID() will not work on a null Object.

I actually don't understand the check for null, how can a channel be null? 

Cheers

Reply:
getThing().getChannel(
The above can return NULL if the channel does not exist. However you have removed the use of this and instead used a constant instead which I don't see the issue in doing it this way. The only thing to comment on is you can use an import and change it to...

```
updateChannelState(CHANNEL_SENEC_POWER_CONSUMPTION, new QuantityType<Power>(.............
```